examples/tree_3d_dgsem/elixir_mhd_diffusive_convergence.jl

```diff
@@ -0,0 +1,148 @@
+using OrdinaryDiffEq
+using Trixi
+
+###############################################################################
+# semidiscretization of the visco-resistive compressible MHD equations
+
+prandtl_number() = 0.72
+mu_const = 0e-2
+eta_const = 0e-2
+
+equations = IdealGlmMhdEquations3D(5 / 3)
+equations_parabolic = ViscoResistiveMhd3D(equations, mu = mu_const,
+                                          Prandtl = prandtl_number(),
+                                          eta = eta_const,
+                                          gradient_variables = GradientVariablesPrimitive())
+
+volume_flux = (flux_hindenlang_gassner, flux_nonconservative_powell)
+solver = DGSEM(polydeg = 3,
+               surface_flux = (flux_lax_friedrichs, flux_nonconservative_powell),
+               volume_integral = VolumeIntegralFluxDifferencing(volume_flux))
+
+coordinates_min = (-1.0, -1.0, -1.0) # minimum coordinates (min(x), min(y), min(z))
+coordinates_max = (1.0, 1.0, 1.0) # maximum coordinates (max(x), max(y), max(z))
+
+# Create a uniformly refined mesh
+mesh = TreeMesh(coordinates_min, coordinates_max,
+                initial_refinement_level = 2,
+                n_cells_max = 200_000) # set maximum capacity of tree data structure
+
+function initial_condition_constant_alfven(x, t, equations)
+    # Alfvén wave in three space dimensions modified by a periodic density variation.
+    # For the system without the density variations see: Altmann thesis http://dx.doi.org/10.18419/opus-3895.
+    # Domain must be set to [-1, 1]^3, γ = 5/3.
+    omega = 2.0 * pi # may be multiplied by frequency
+    # r = length-variable = length of computational domain
+    r = 2.0
+    # e = epsilon
+    e = 0.02
+    nx = 1 / sqrt(r^2 + 1.0)
+    ny = r / sqrt(r^2 + 1.0)
+    sqr = 1.0
+    Va = omega / (ny * sqr)
+    phi_alv = omega / ny * (nx * (x[1] - 0.5 * r) + ny * (x[2] - 0.5 * r)) - Va * t
+
+    rho = 1.0 + e*cos(phi_alv + 1.0)
+    v1 = -e * ny * cos(phi_alv) / rho
+    v2 = e * nx * cos(phi_alv) / rho
+    v3 = e * sin(phi_alv) / rho
+    p = 1.0
+    B1 = nx - rho * v1 * sqr
+    B2 = ny - rho * v2 * sqr
+    B3 = -rho * v3 * sqr
+    psi = 0.0
```

Comment:
This looks like a fairly complicated manufactured solution that would be the ideal MHD Alfvén wave (so no source term required if viscosity and resistivity are set to 0). If I may propose an alternative, here is an MMS from my old (2D) Fortran code:
```fortran
            alpha = 2.0_RP*PI*(x+y)-4.0_RP*t
            Resu(1) = SIN(alpha)+4.0_RP
            Resu(2) = SIN(alpha)+4.0_RP
            Resu(3) = SIN(alpha)+4.0_RP
            Resu(4) = 0.0_RP
            Resu(5) = 2.0_RP*(SIN(alpha)+4.0_RP)**2
            Resu(6) = SIN(alpha)+4.0_RP
            Resu(7) = -SIN(alpha)-4.0_RP
            Resu(8) = 0.0_RP
            Resu(9) = 0.0_RP
```

The associated source term is then (fairly) simple and takes the form:
```fortran
            alpha = 2.0_RP*PI*(x+y)-4.0_RP*t
            phi = SIN(alpha)+4.0_RP
            phi_t = -4.0_RP*COS(alpha)
            phi_x = 2.0_RP*PI*COS(alpha)
            phi_xx = -4.0_RP*PI**2*SIN(alpha)
            source(1) = phi_t+2.0_RP*phi_x
            source(2) = phi_t+4.0_RP*phi*phi_x+phi_x
            source(3) = source(2)
            source(4) = 0.0_RP
            source(5) = 4.0_RP*phi*phi_t+16.0_RP*phi*phi_x-2.0_RP*phi_x-4.0_RP*eta*(phi_x**2+phi*phi_xx)-4.0_RP*mu/Prandtl*phi_xx
            source(6) = source(1)-2.0_RP*eta*phi_xx
            source(7) = -source(6)
            source(8) = 0.0_RP
            source(9) = 0.0_RP
```
Maybe a somewhat simpler MMS for the 3D case could help in this instance.

Reply:
What are domain and final time for this case? Domain probably $[0, 1]^2$?

Reply:
Yes, sorry, the domain in [0,1]^2 and the final time I typically used 1.5. The `Resu` is already the conservative variables such that we are setting the velocities to be `(v1_, v_2, v_3) = (1, 1, 0)`.

Reply:
Thanks!

Reply:
Thanks for that. I had a go with this and tried some modifications, but the convergence rate is close to 0, even without viscosity or magnetic resistivity. I am running to time 0.1, though, as it takes rather long.